contrib/gdevsvg.c

```diff
@@ -389,7 +389,19 @@ svg_endpath(gx_device_vector* vdev, gx_path_type_t type);
 /* structure to store PNG image bytes */
 struct mem_encode
 {
-	char* buffer;
+	// TRS 03 July 2024
+	// Note: The buffer is currently 128KB. This value was chosen becuase 256KB 
+	// crashed Ghostscript and so 128KB was used. No values between 128KB and 
+	// 256KB were tested. Smaller values might work as well, but if we ever 
+	// receieve data with a chunk size greater than this buffer's length in the
+	// my_png_write_data function, we'll crash the program. As such, we want to
+	// use the largest buffer for this as possible.
+	// Note: Originally, buffer was dynamically allocated using gs_resize_object,
+	// grown using gs_resize_object, and freed using gs_free_object. That code 
+	// crashed for reasons I was unable to determine. While the new code has 
+	// the disadvantages described above, it is at least simpler and easier to 
+	// read.
+	char buffer[1024 * 128];
```

Comment:
Should we dynamically allocate this instead?

Reply:
It looks like that is what was happening in the past? `gs_alloc_bytes` call later that was removed?

Reply:
Yes, that's what we were doing, but as the comment says, it caused GS to crash and I was unable to determine why. This new method does not crash using the files that I tested with.